packages/flutter_test/lib/src/controller.dart

```diff
@@ -95,6 +104,104 @@ class SemanticsController {
     return result;
   }
 
+  /// Attemps to find a [SemanticsNode] with the given `action` as close to the
+  /// [SemanticsNode] of `finder` as possible.
+  ///
+  /// The [SemanticsNode] directly related to an [Element] is not necessarily
+  /// the one that can be acted upon. This method finds the directly related
+  /// node and checks whether it supports the given `action`. If the directly
+  /// related node does not support the given `action`, then the closest
+  /// descendent of that node that does support the `action` is returned.
+  ///
+  /// Will throw a [StateError] if:
+  /// * No results are returned from `finder`
+  /// * More than result is returned from `finder`
+  /// * The [SemanticsNode] of `finder` or its descendant nodes do not support
+  ///   `action`
+  ///
+  /// See also:
+  /// * [find]
+  /// * [findWithAnyAction]
+  SemanticsNode findWithAction(Finder finder, SemanticsAction action) {
+    final SemanticsNode found = find(finder);
+    SemanticsNode? actionable;
+    _traverse(found, (SemanticsNode node) {
+      if (node.getSemanticsData().hasAction(action)) {
+        actionable = node;
+        return true;
+      }
+      return false;
+    });
+
+    if (actionable == null) {
+      throw StateError(
+        'The given finder did not find a node that supports (or has '
+        'a descendant that supports) $action.\n'
+        'Found Node: $found'
+      );
+    }
+    return actionable!;
+  }
+
+  /// Attemps to find a [SemanticsNode] with any of the given `actions` as
+  /// close to the [SemanticsNode] of `finder` as possible.
+  ///
+  /// The [SemanticsNode] directly related to an [Element] is not necessarily
+  /// the one that can be acted upon. This method finds the directly related
+  /// node and checks whether it supports any of the given `actions`. If the
+  /// directly related node does not support any of the given `actions`, then
+  /// the closest descendent of that node that does support at least one of
+  /// the `actions` is returned.
+  ///
+  /// Will throw a [StateError] if:
+  /// * No results are returned from `finder`
+  /// * More than result is returned from `finder`
+  /// * The [SemanticsNode] of `finder` or its descendant nodes do not support
+  ///   at least one action in `actions`
+  ///
+  /// See also:
+  /// * [find]
+  /// * [findWithAction]
+  SemanticsNode findWithAnyAction(Finder finder, List<SemanticsAction> actions) {
+    final int actionInt = actions.fold<int>(0,
+      (int prev, SemanticsAction act) => prev | act.index);
```

Comment:
nit: formatting, I would probably leave this all on one line, if you want to split it over multiple, the line with the `(` and the line with the `)` should have the same indentation, so you would end up with something like

```dart
    final int actionInt = actions.fold<int>(
      0,
      (int prev, SemanticsAction act) => prev | act.index,
    );
```

Reply:
I prefer to split it over the multiple lines so the actual "folding" action is explicit and easy to see. TBH, it'd be nice to not have to specify the types in situations like these...

I did update it to have the closing parenthesis on its own line though.